dask/dataframe/indexing.py

```diff
@@ -133,6 +136,13 @@ def _loc_series(self, iindexer, cindexer):
         return self.obj.map_partitions(methods.loc, iindexer, cindexer,
                                        token='loc-series', meta=meta)
 
+    def _loc_array(self, iindexer, cindexer):
+        iindexer_series = Series(iindexer.dask,
+                                 iindexer.name,
+                                 pd.Series([], dtype=bool),
+                                 self.obj.divisions)
```

Comment:
I wonder if the `to_dask_dataframe` method might be more robust here.

Reply:
tried it, works.